test/will.cpp

```diff
@@ -572,7 +574,7 @@ BOOST_AUTO_TEST_CASE( overlength_message ) {
     c1->set_client_id("cid1");
     c1->set_clean_session(true);
     std::string wm(0x10000, 'a');
-    c1->set_will(mqtt::will("topic1", wm));
+    c1->set_will(mqtt::will("topic1"_mb, mqtt::buffer(mqtt::string_view(wm))));
```

Comment:
In this test, https://github.com/redboltz/mqtt_cpp/pull/339/files#diff-1d10869b5251c92570e5c05a9f9a643eR581 keeps lifetime of the wm (on the stack).
I think that it is safe, but could be misleading message for someone refer to the code.

I think that **explicit** allocate buffer function is helpful.

in buffer.hpp

```cpp
inline buffer allocate_buffer(string_view sv) {
    auto spa = make_shared_ptr_array(sv.size());
    std::copy(sv.begin(), sv.end(), spa.get());
    return buffer(string_view(spa.get(), spa.get() + sv.size()), spa);
}

inline buffer allocate_buffer(char const* ptr, std::size_t size) {
    return allocate_buffer(string_view(ptr, size));
}
```

It can be used other places.

Reply:
https://github.com/redboltz/mqtt_cpp/pull/339/files#r315478183 is better. I use it.

Reply:
Ok, that's fine with me.

Reply:
done(just my memo, not pushed yet).